test/unix/io_array.ml

```diff
@@ -19,15 +19,21 @@ module Entry = struct
     (key, value)
 
   let append_io io (key, value) =
-    let encoded_key = Key.encode key in
-    let encoded_value = Value.encode value in
-    IO.append io encoded_key;
-    IO.append io encoded_value
+    (* Even though the key is an unary value, if a crash occurs between the
+       [Key.encode] and [Value.encode] calls, the flushed value will be
+       incomplete. Hence the use of [append_batch] *)
+    IO.append_batch io (fun f ->
+        Key.encode key f;
+        Value.encode value f)
 end
 
 module IOArray = Index.Private.Io_array.Make (IO) (Entry)
 
-let entry = Alcotest.(pair string string)
+let key = Alcotest.(string)
+
+let value = Alcotest.(pair string string)
+
+let entry = Alcotest.(pair key value)
```

Comment:
why introducing the extra `key` and `value` here?

Reply:
Because I was afraid I would have to add these types everywhere and I didn't want to write them multiple time
The other reason was that if you ever change one of these types in these tests you just need to edit these lines